src/bootstrap/native.rs

```diff
@@ -916,7 +916,51 @@ impl Step for Lld {
         t!(fs::create_dir_all(&out_dir));
 
         let mut cfg = cmake::Config::new(builder.src.join("src/llvm-project/lld"));
-        configure_cmake(builder, target, &mut cfg, true, LdFlags::default());
+        let mut ldflags = LdFlags::default();
+
+        // When building LLD as part of a build that asks LLVM to generate profiles, for example
+        // when doing PGO on CI, something on windows (maybe `clang-cl`) doesn't automatically link
+        // the profiler runtime in. We manually link the lib in that case to avoid linking errors:
+        // it's also located in the same precompiled LLVM distribution as `clang-cl`.
+        //
+        // Since we only do PGO on CI for x64 windows, handle that specifically, only for the config
+        // the bots use on CI (like we do for building llvm itself).
+        if builder.config.llvm_profile_generate
+            && target.contains("msvc")
+            && target.contains("x86_64")
+            && builder.ninja()
+            && builder.config.ccache.is_some()
```

Comment:
I'm not sure about this condition. It might be too specific, that is, if we specify `llvm_profile_generate`, we are building an LLVM that will need `clang_rt.profile-x86_64.lib`, but here we only add it to the linker invocation for a subset of those cases. So we might get linker errors in those cases (e.g. if someone tries to build locally but e.g. not with ninja).

Maybe the following would work?:
```rust
if builder.config.llvm_profile_generate {
    if let Some(clang_cl_path) = builder.config.llvm_clang_cl.as_ref() {
        // ...
    }
}
```

Reply:
I haven't fully tested all the options, so I still locally test similarly to CI, but I have at least made the config only check for the clang_cl path on an MSVC target.